src/lib/select/option.ts

```diff
@@ -1,10 +1,82 @@
-import {Component, ViewEncapsulation} from '@angular/core';
+import {
+  Component,
+  ElementRef,
+  EventEmitter,
+  Output,
+  Renderer,
+  ViewEncapsulation
+} from '@angular/core';
+import {ENTER, SPACE} from '../core/keyboard/keycodes';
 
 @Component({
   moduleId: module.id,
   selector: 'md-option',
-  template: ``,
+  host: {
+    'role': 'option',
+    '[class.md-selected]': 'selected',
+    '[attr.aria-selected]': 'selected.toString()',
+    '[attr.tabindex]': '0',
+    '(click)': 'select()',
+    '(keydown)': '_selectOnActivate($event)'
+  },
+  template: `
+    <ng-content></ng-content>
+    <div class="md-option-ripple" md-ripple md-ripple-background-color="rgba(0,0,0,0)" 
+      [md-ripple-trigger]="_getHostElement()"></div>
+   `,
   styleUrls: ['select.css'],
   encapsulation: ViewEncapsulation.None
 })
-export class MdOption {}
+export class MdOption {
+  private _selected = false;
+
+  /** Event emitted when the option is selected. */
+  @Output() onSelect = new EventEmitter();
+
+  constructor(private _element: ElementRef, private _renderer: Renderer) {}
+
+  /** Whether or not the option is currently selected. */
+  get selected() {
+    return this._selected;
+  }
+
+  /**
+   * The displayed value of the option.
+   * TODO(kara): Add input property alternative for node envs.
+   */
+  get viewValue(): string {
+    return this._getHostElement().textContent.trim();
+  }
+
+  /** Selects the option. */
+  select() {
+    this._selected = true;
+    this.onSelect.emit();
+  }
+
+  /** Unselects the option. */
+  unselect() {
+    this._selected = false;
+  }
+
+  /** Sets focus onto this option. */
+  focus() {
+    this._renderer.invokeElementMethod(this._getHostElement(), 'focus');
+  }
+
+  /**
+   * Ensures the option is selected when activated from the keyboard.
+   * TODO: internal
```

Comment:
Let's stop adding new instances of `TODO: internal`. At this point we're mostly settled into the convention of underscore without `private` as meaning internal. 
(we can remove the old ones in a cleanup pass)